### DISCUSSION TITLE
How to set default value of Primary Key in MySQL Bridge Table

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#6567] [Help]
I am having an issue when selecting values in a many-to-many relationship picker and then saving the form, the form table entry will get saved, however the relationship in the bridge table will not. The error that appears is that 'Field "id" doesn't have a default value', where 'id' is my primary key in the bridge table. My standard practice is to concatenate the two foreign values stored in the bridge table and use it as the primary key. It appears that within Budibase I can't just declare the primary key to be made up of those two columns.

The question, is there a way in Budibase MySQL datasources to actually declare a default value for a column by using concatenation of other columns in the same table when there is no form to be able to do this otherwise? Is the only alternative to set up a routine in the MySQL database for the onbeforecreate event otherwise?

**Comment by melohagan** [comment]
Hi @mmalessa80 

If I understand correctly, you are looking for composite key support?

There is an issue open here: https://github.com/Budibase/budibase/issues/1738

**Reply by mmalessa80** [reply]
Yes, I didn't realize there was a request for this already. Alternatively if there was a way to set a default value through the relationship picker, that would work also. Or, I am attempting to implement automation with looping instead to just create the rows in the bridge table, but I ran into problems as explained here https://github.com/Budibase/budibase/discussions/6713.
